control_panel_api/serializers.py

```diff
@@ -161,3 +145,43 @@ def update(self, instance, validated_data):
             )
 
         return super().update(instance, validated_data)
+
+
+class UserUserAppSerializer(serializers.ModelSerializer):
```

Comment:
This class name is not very self-documenting. Should it be something like `NestedUserAppSerializer`?

Reply:
I did originally name it Nested* but then I have another one so can't call it Nested2* so I prefixed the name of the Serializer that owns it.

So UserUserAppSerializer means the serializer for UserApps on User.  If that makes sense.

Reply:
👍

Reply:
As showed offline, I was playing with the names of the serializers and I came up with something which may be a bit more explicit.

Here the rules:
1. Serialiser class start with name of the model serialised (e.g. `App...` is serialising an `App`, `AppS3Bucket...` is serialising an `AppS3Bucket`
2. If the serialiser is not serialising relationships we prefix it with `...Simple` (e.g. `AppSimple...` is serializing an `App` without associations, etc...)
3. If the serialiser is going to be used/nested in another serialiser will have prefix `...NestedIn${ParentSerializerName}` (e.g. `AppS3BucketNestedInAppSerializer` is the serialiser for `AppS3Bucket` in the context of an `App`, `AppS3BucketNestedInS3BucketSerializer` is the serialiser for `AppS3Bucket` in the context of an `S3Bucket`)

Examples:
* `UserUserAppSerializer` => `UserAppNestedInAppSerializer`
* `UserUserS3BucketSerializer` => `UserS3BucketNestedInUserSerializer`
* `SimpleS3BucketSerializer` => `S3BucketSimpleSerializer`
* etc...

Benefits:
* Explicit, the name of the class tells you where the serialiser will be nested/used.
* Class names start with the object being serialized

Reply:
I like it, except for `...Simple` - that seems unnecessary

Reply:
It's called simple as it just exposes some of the top level S3Bucket fields, as opposed to the normal S3BucketSerializer which also exposes the many to many and also displays the m2m as a nested representation.

Reply:
Alternatives to `...Simple`:
 * `...NoAssociations`

Reply:
@jmoz so basically it's the serialisation of the core fields but it doesn't contain the associations (as the purpose of these "simple" serializers is to be used in the context of the parents associations so they're essentially "leaves" in the JSON response "tree").

The more I think about the `...NoAssociations` name the more I like it as it's quite explicit.

Reply:
AppFLATSerializer!?